src/BPEssentials/Core.cs

```diff
@@ -159,6 +160,24 @@ public void RegisterCustomCommands()
             }
         }
 
+        public void DeleteExpiredWarns()
+        {
+            foreach (var user in Instance.SvManager.database.Users.FindAll())
+            {
+                if (!user.Character.CustomData.TryFetchCustomData<List<SerializableWarn>>(CustomDataKey, out var warns))
+                {
+                    continue;
+                }
+                foreach (var warn in warns)
+                {
+                    if (warn.Date.AddDays(warn.Length) <= DateTime.Now)
+                    {
+                        warns.Remove(warn);
+                    }
+                }
+            }
+        }
+
```

Comment:
Should we delete them or mark them as expired?

Reply:
I can give that a setting